source/NVDAObjects/UIA/__init__.py

```diff
@@ -1676,12 +1676,18 @@ def event_UIA_systemAlert(self):
 	def event_UIA_notification(self, notificationKind=None, notificationProcessing=UIAHandler.NotificationProcessing_CurrentThenMostRecent, displayString=None, activityId=None):
 		"""
 		Introduced in Windows 10 Fall Creators Update (build 16299).
-		This base implementation announces all notifications from the UIA element.
+		This base implementation announces all notifications from the UIA element if enabled from NVDA.
 		Unlike other events, the text to be announced is not the name of the object, and parameters control how the incoming notification should be processed.
 		Subclasses can override this event and can react to notification processing instructions.
 		"""
 		# Do not announce notifications from background apps.
-		if self.appModule != api.getFocusObject().appModule:
+		# #10956: ignore this altogether if NVDA should not handle this event
+		# either always or from background apps.
+		reportUIANotifications = config.conf['presentation']['reportUIANotifications']
+		if (
+			reportUIANotifications == "off"
+			or reportUIANotifications == "focusedApp" and self.appModule != api.getFocusObject().appModule

```

Comment:
Could you add parenthesis here to make this clearer to read. It looks like this does what you intended but I had to double check the operator precedence to be sure (for `and`, `or` operators)
```suggestion
			or (
				reportUIANotifications == "focusedApp"
				and self.appModule != api.getFocusObject().appModule
			)

```